LibreNMS/OS/IbmImm.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * IbmImm.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       https://www.librenms.org
+ * @copyright  2025 Peca Nesovanovic
+ * @author     Peca Nesovanovic <peca.nesovanovic@sattrakt.com>
+ */
+
+namespace LibreNMS\OS;
+
+use App\Models\Device;
+use LibreNMS\Interfaces\Data\DataStorageInterface;
+use LibreNMS\Interfaces\Discovery\OSDiscovery;
+use LibreNMS\Interfaces\Polling\OSPolling;
+use SnmpQuery;
+
+class IbmImm extends \LibreNMS\OS implements OSDiscovery, OSPolling
+{
+    public function discoverOS(Device $device): void
+    {
+        parent::discoverOS($device); // yaml
+
+        $this->customSysName($device);
+
+        $oids = SnmpQuery::hideMib()->walk('IMM-MIB::immVpdType')->table();
+        foreach ($oids['immVpdType'] ?? [] as $key => $type) {
+            $device->features .= ' ' . $type;
+        }
+    }
+
+    public function pollOS(DataStorageInterface $datastore): void
+    {
+        $this->customSysName($this->getDevice());
+    }
+
+    private function customSysName(Device $device): void
+    {
+        $device->sysName = SnmpQuery::get('IMM-MIB::spTxtId.0')->value() ?: $device->sysName;
```

Comment:
This can be done in yaml.

Reply:
> can be done in yaml.

sorry @murrant 
i dont see any `sysname:` or similar in any existing YAML
tried with sysname: sysName: hostname: hosName: etc ... but no, YAML discovery wont pick up sysname

instead, i see in many OS php there is `customSysName` function for this reason

any idea how to grab sysname from YAML ?

Reply:
ok @murrant 
i added `sysname:` field to YAML discovery
after all, this will prevent constant eventlog flooding with messages about sysname changes
... if you agree